### PR TITLE
build: Latest go-mods, config and Dockerfile fix

### PR DESCRIPTION
**Pull request by lenny-intel**
Signed-off-by: Leonard Goodell <leonard.goodell@intel.com>

**If your build fails**  due to your commit message not passing the build checks, please review the guidelines here: https://github.com/edgexfoundry/edgex-go/blob/main/.github/Contributing.md

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] I am not introducing a breaking change (if you are, flag in conventional commit message with `BREAKING CHANGE:` describing the break)
- [x] I am not introducing a new dependency (add notes below if you are)
- [ ] I have added unit tests for the new feature or bug fix (if not, why?) **N/A**
- [ ] I have fully tested (add details below) this the new feature or bug fix (if not, why?) **N/A**
- [ ] I have opened a PR for the related docs change (if not, why?) **N/A**
      <link to docs PR>

### CODE REVIEW

File: cmd/core-data/Dockerfile [hunk 1]
```diff
@@ -27,8 +27,6 @@ WORKDIR /edgex-go
 # The main mirrors are giving us timeout issues on builds periodically.
 # So we can try these.
 
-RUN sed -e 's/dl-cdn[.]alpinelinux.org/dl-4.alpinelinux.org/g' -i~ /etc/apk/repositories
-
```
Comment (jim-wang-intel) [inline cmd/core-data/Dockerfile:31]: also remove the comments

File: cmd/core-command/Dockerfile [hunk 2]
```diff
@@ -26,8 +26,6 @@ WORKDIR /edgex-go
 # The main mirrors are giving us timeout issues on builds periodically.
 # So we can try these.
 
-RUN sed -e 's/dl-cdn[.]alpinelinux.org/dl-4.alpinelinux.org/g' -i~ /etc/apk/repositories
-
```
Comment (jim-wang-intel) [inline cmd/core-command/Dockerfile:30]: also remove the comments

File: cmd/core-metadata/Dockerfile [hunk 3]
```diff
@@ -26,8 +26,6 @@ WORKDIR /edgex-go
 # The main mirrors are giving us timeout issues on builds periodically.
 # So we can try these.
 
-RUN sed -e 's/dl-cdn[.]alpinelinux.org/dl-4.alpinelinux.org/g' -i~ /etc/apk/repositories
-
```
Comment (jim-wang-intel) [inline cmd/core-metadata/Dockerfile:30]: also remove the comments

File: cmd/support-notifications/Dockerfile [hunk 4]
```diff
@@ -25,8 +25,6 @@ WORKDIR /edgex-go
 # The main mirrors are giving us timeout issues on builds periodically.
 # So we can try these.
 
-RUN sed -e 's/dl-cdn[.]alpinelinux.org/dl-4.alpinelinux.org/g' -i~ /etc/apk/repositories
-
```
Comment (jim-wang-intel) [inline cmd/support-notifications/Dockerfile:29]: also remove the comments

File: cmd/support-scheduler/Dockerfile [hunk 5]
```diff
@@ -26,8 +26,6 @@ WORKDIR /edgex-go
 # The main mirrors are giving us timeout issues on builds periodically.
 # So we can try these.
 
-RUN sed -e 's/dl-cdn[.]alpinelinux.org/dl-4.alpinelinux.org/g' -i~ /etc/apk/repositories
-
```
Comment (jim-wang-intel) [inline cmd/support-scheduler/Dockerfile:30]: also remove the comments

File: cmd/core-data/Dockerfile (hunk 1 above)
Comment (lenny-intel) [reply cmd/core-data/Dockerfile:31]: yep, did on some, missed on these . Good catch.

**Review by ernestojeda — APPROVED** [review]

File: cmd/core-data/Dockerfile (hunk 1 above)
Comment (lenny-intel) [reply cmd/core-data/Dockerfile]: Removed all left over comments

**Review by jim-wang-intel — APPROVED** [review]

**Review by bill-mahoney — APPROVED** [review]

**Pull request merged**